org.eclipse.lsp4e/src/org/eclipse/lsp4e/command/CommandExecutor.java

```diff
@@ -64,23 +64,25 @@ public class CommandExecutor {
 	private static final String LSP_COMMAND_PARAMETER_TYPE_ID = "org.eclipse.lsp4e.commandParameterType"; //$NON-NLS-1$
 	private static final String LSP_PATH_PARAMETER_TYPE_ID = "org.eclipse.lsp4e.pathParameterType"; //$NON-NLS-1$
 
-	public static CompletableFuture<Object> executeCommandClientSide(@NonNull Command command, @NonNull IDocument document) {
-		IFile file = LSPEclipseUtils.getFile(document);
-		if (file != null) {
-			return CommandExecutor.executeCommandClientSide(command, file);
-		}
-		return null;
+	/**
+	 * @deprecated use {@link #executeCommandClientSide(Command, IDocument)}
+	 */
+	@Deprecated(forRemoval = true)
+	public static CompletableFuture<Object> executeCommand(@Nullable Command command, @Nullable IDocument document,
```

Comment:
If we restore this method, I think it should also do the server request, otherwise we keep the API stable but it does not do the same as before, which I think it is more error prone than the breaking change + asking consumer to refactor. Since I broke it, If you want that I can do that, but given that the class was deprecated already and that we want to promote the clean pattern which the old code did not support, I would be rather in favor of breaking the API.

Reply:
I agree with it; but we can proceed as you prefer: either we merge it and then re-add server-side execution; or we try to fix it before merging, or we decide it's not that critical to support server-side command executions here. FWIW, I checked on GitHub search and in the workspace, and I think all consumers I've noticed do assume it's client-side, so loosing server-side commands here will not be perceived as a regression by those, ie there is most likely no actual problem to solve...

Reply:
If all consumers assume it's client-side, then what about just https://github.com/eclipse/lsp4e/pull/506?

Reply:
This will restore the method indeed. But we also need the command execution to work with non-IResource document, as I believe it used to (although I'm not 100% sure) and to be able to generate a workspaceEdit from a URI without requiring an IResource.

Reply:
Okay, I will look at how to restore that behavior as well, later. I think it should be a small adaptation to #506